config/locales/fr.yml

```diff
@@ -35,6 +35,7 @@ fr:
     year: Anée
   date:
     abbr_month_names:
+    -
```

Comment:
month 0 doesn't exist, the key is needed, see https://github.com/rails/rails/blob/d59e729fbf48858b672814194a180d6ddf795f3d/activesupport/lib/active_support/locale/en.yml#L16